src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php

```diff
@@ -178,4 +525,300 @@ public function thePromotionShouldNotAppliesToDiscountedItems(PromotionInterface
+                $usageLimit,
+            ),
+        );
+    }
+
+    /**
+     * @Then the :promotion promotion should be exclusive
+     */
+    public function thePromotionShouldBeExclusive(PromotionInterface $promotion): void
+    {
+        Assert::true(
+            $this->responseChecker->getValue(
+                $this->client->show(Resources::PROMOTIONS, $promotion->getCode()),
+                'exclusive',
+            ),
+        );
+    }
+
+    /**
+     * @Then the :promotion promotion should be coupon based
+     */
+    public function thePromotionShouldBeCouponBased(PromotionInterface $promotion): void
+    {
+        Assert::true(
+            $this->responseChecker->getValue(
+                $this->client->show(Resources::PROMOTIONS, $promotion->getCode()),
+                'couponBased',
+            ),
+        );
+    }
+
+    /**
+     * @Then the :promotion promotion should be applicable for the :channel channel
+     */
+    public function thePromotionShouldBeApplicableForTheChannel(PromotionInterface $promotion, ChannelInterface $channel): void
+    {
+        Assert::true(
+            $this->responseChecker->hasValueInCollection(
+                $this->client->show(Resources::PROMOTIONS, $promotion->getCode()),
+                'channels',
+                $this->sectionAwareIriConverter->getIriFromResourceInSection($channel, 'admin'),
+            ),
+            sprintf('Catalog promotion is not assigned to %s channel', $channel->getName()),
+        );
+    }
+
+    /**
+     * @When the :promotion promotion should have a label :label in :localeCode locale
+     */
+    public function thePromotionShouldHaveLabelInLocale(PromotionInterface $promotion, string $label, string $localeCode): void
+    {
+        $response = $this->client->show(Resources::PROMOTIONS, $promotion->getCode());
+
+        Assert::true($this->responseChecker->hasTranslation($response, $localeCode, 'label', $label));
+    }
+
+    /**
+     * @Then /^it should have ("[^"]+") of item percentage discount configured for ("[^"]+" channel)$/
+     */
+    public function itShouldHaveOfItemPercentageDiscount(float $percentage, ChannelInterface $channel): void
+    {
+        $actions = $this->responseChecker->getValue($this->client->getLastResponse(), 'actions');
+        foreach ($actions as $action) {
+            if ($action['type'] === 'unit_percentage_discount') {
+                Assert::same((float) $action['configuration'][$channel->getCode()]['percentage'], $percentage);
+            }
+        }
+    }
+
+    /**
+     * @Then /^it should have ("[^"]+") of order percentage discount$/
+     */
+    public function itShouldHaveOfOrderPercentageDiscount(float $percentage): void
+    {
+        $actions = $this->responseChecker->getValue($this->client->getLastResponse(), 'actions');
+        Assert::same((float) $actions[0]['configuration']['percentage'], $percentage);
+    }
+
+    /**
+     * @Then the code field should be disabled
```

Comment:
```suggestion
     * @Then I should not be able to edit its code
```
and adjust the scenario